### PR TITLE
Split out list service

### PR DESCRIPTION
**Pull request by alex-deaconu**
Using it for presentations and products lists

[stage-2]

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc @rodrigopavezi can you please take a look at this PR?

Basically, I've created a service that facilitates Scrolling List functionality for any controller. It will create the search object with default values (or receive one from the controller), and provide all the list functions that the controller needs: load, search and sort.

Pretty much prevents duplicate code for lists.

Please let me know if you have any comments or suggestions on how to improve this.

Thanks!

File: js/services/svc-scrolling-list-service.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            factory.loadingItems = true;
+
+            listService(factory.search, factory.items.cursor)
+              .then(function (result) {
+                factory.items.add(result.items ? result.items : [],
+                  result.cursor);
+              })
+              .then(null, function (e) {
+                factory.error =
+                  'Failed to load presentations. Please try again later.';
```
Comment (ezequielc) [inline js/services/svc-scrolling-list-service.js]: @alex-deaconu This message should not refer to presentations, right?

File: js/services/svc-scrolling-list-service.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    function (BaseList) {
+      return function (listService, search) {
+        var DB_MAX_COUNT = 40; //number of records to load at a time
+        var factory = {};
+
+        factory.items = new BaseList(DB_MAX_COUNT);
+
+        factory.search = search ? search : {};
+        _.defaults(factory.search, {
+          sortBy: 'changeDate',
```
Comment (ezequielc) [inline js/services/svc-scrolling-list-service.js]: @alex-deaconu what happens if the item does not have changeDate property?

**Comment by ezequielc** [comment]
@alex-deaconu Just two minor comments. Looks good. :+1:

File: js/services/svc-scrolling-list-service.js (hunk 1 above)
Comment (alex-deaconu) [inline js/services/svc-scrolling-list-service.js]: @ezequielc good catch, will fix. Thanks.

File: js/services/svc-scrolling-list-service.js (hunk 2 above)
Comment (alex-deaconu) [inline js/services/svc-scrolling-list-service.js]: @ezequielc good question. From an API standpoint, nothing should happen, it should just ignore the sort parameter. Do you have any suggestions on how I could deal with this case?

However, you bring up a good point. Generally lists are either sorted by `changeDate` or `name`. I just assumed `changeDate` is the more common of the two, so I added it here as default. Do you think I should default to `name`? 

Of course, the value can be overwritten by providing a `search` object.

Thanks!

**Comment by rodrigopavezi** [comment]
@alex-deaconu LG :+1:

File: js/services/svc-scrolling-list-service.js (hunk 2 above)
Comment (ezequielc) [inline js/services/svc-scrolling-list-service.js]: @alex-deaconu Not sure also. One idea would be to leave `search` empty and get the default sort from the endpoint API. 
But as you said, the value can be overwritten, so it is not a big problem. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc unfortunately if the sort is missing or using a wrong column name, the list returns in a very random order. I changed the sort by default to `name`, this column should be there most of the time. Also fixed the typo. Thanks.

**Pull request merged**